src/libPersistence/ContractStorage2.cpp

```diff
@@ -160,15 +168,19 @@ bool ContractStorage2::FetchStateValue(const dev::h160& addr, const bytes& src,
     if (t_type.empty()) {
       LOG_GENERAL(WARNING, "Failed to fetch type for addr: "
                                << addr.hex() << " vname: " << query.name());
-      return false;
+      foundVal = false;
+      return true;
     }
     try {
       type = DataConversion::CharArrayToString(t_type[type_key]);
     } catch (const std::exception& e) {
       LOG_GENERAL(WARNING, "Invalid type fetched for key="
                                << type_key << " for addr=" << addr.hex() << ": "
                                << e.what());
+      return false;
     }
+    // If not interested in the value, exit early.
+    if (query.indices().empty() && query.ignoreval()) return true;
```

Comment:
`query.indices().empty()` implies that we are only looking at a state variable (field) as a whole, not indexing into a map (if the state variable is a map).

In case the indices list is not empty, that means we're checking to see if a (nested) map key exists (and depending on whether `ignoreval` is set then, the value for that key may be needed too).

In this case, if the indices list is empty, we only care whether the field itself exists (and its type) or not.